src/script/interpreter.cpp

```diff
@@ -1030,6 +1032,7 @@ class CTransactionSignatureSerializer {
         // Serialize the prevout
         ::Serialize(s, txTo.vin[nInput].prevout, nType, nVersion);
         // Serialize the script
+        assert(nInput != NOT_AN_INPUT);
```

Comment:
I'm not sure if `CheckBlock` / `AcceptToMemoryPool` triggers this code, in which case this assertion could be abused, so let's double-check.